### ISSUE TITLE
Node 1.0.3 stops to send proposal because of NATS at some point

### ISSUE THREAD

**Issue by soffokl** [mysteriumnetwork/node#4402]
Node restart resolve the problem.

```
Dec 01 22:30:34 ubuntu myst[1219162]: 2021-12-01T22:30:34.183 ERR core/discovery/discovery.go:208          > Failed to ping proposal error="failed to ping proposal: {0 service-proposal/v3 1 0x05b9004d70869f0c2e3da5f04d8229b89948b655 openvpn {NA CA Unknown 31798 DATACITY hosting} [{nats/p2p/v1 {[nats://broker.mysterium.network:4222 nats://broker.mysterium.network:4222 nats://51.158.129.204:4222]}}] <nil> {0 0 0}}: failed to send message 'signed.fOwEcqDkuW17eXkfDLNq7FGG3dhJftmrkRj_lOIW7U5DQeSmKDL_vMFg4Xy8TYUmS39ZkTCwrdkKgwA-GYd6HwE.1638415834.*.proposal-ping.v3': nats: connection closed"
Dec 01 22:30:34 ubuntu myst[1219162]: 2021-12-01T22:30:34.183 ERR core/discovery/discovery.go:208          > Failed to ping proposal error="failed to ping proposal: {0 service-proposal/v3 1 0x05b9004d70869f0c2e3da5f04d8229b89948b655 wireguard {NA CA Unknown 31798 DATACITY hosting} [{nats/p2p/v1 {[nats://broker.mysterium.network:4222 nats://broker.mysterium.network:4222 nats://51.158.129.204:4222]}}] <nil> {0 0 0}}: failed to send message 'signed.fOwEcqDkuW17eXkfDLNq7FGG3dhJftmrkRj_lOIW7U5DQeSmKDL_vMFg4Xy8TYUmS39ZkTCwrdkKgwA-GYd6HwE.1638415834.*.proposal-ping.v3': nats: connection closed"
Dec 01 22:33:34 ubuntu myst[1219162]: 2021-12-01T22:33:34.184 DBG communication/nats/sender.go:59          > Message "signed.NZYt4_zSpfRzDl-x3KmpGfgrsVZ69Ky7xQyon9Q5CaMUAaBGPFVUxUb02RgdsJ5Qyfl6_64dX8rp3Aou3Cp11wA.1638416014.*.proposal-ping.v3" sending: {"proposal":{"id":0,"format":"service-proposal/v3","compatibility":1,"provider_id":"0x05b9004d70869f0c2e3da5f04d8229b89948b655","service_type":"openvpn","location":{"continent":"NA","country":"CA","city":"Unknown","asn":31798,"isp":"DATACITY","ip_type":"hosting"},"contacts":[{"type":"nats/p2p/v1","definition":{"broker_addresses":["nats://broker.mysterium.network:4222","nats://broker.mysterium.network:4222","nats://51.158.129.204:4222"]}}],"quality":{"quality":0,"latency":0,"bandwidth":0}}}
Dec 01 22:33:34 ubuntu myst[1219162]: 2021-12-01T22:33:34.186 DBG communication/nats/sender.go:59          > Message "signed.NZYt4_zSpfRzDl-x3KmpGfgrsVZ69Ky7xQyon9Q5CaMUAaBGPFVUxUb02RgdsJ5Qyfl6_64dX8rp3Aou3Cp11wA.1638416014.*.proposal-ping.v3" sending: {"proposal":{"id":0,"format":"service-proposal/v3","compatibility":1,"provider_id":"0x05b9004d70869f0c2e3da5f04d8229b89948b655","service_type":"wireguard","location":{"continent":"NA","country":"CA","city":"Unknown","asn":31798,"isp":"DATACITY","ip_type":"hosting"},"contacts":[{"type":"nats/p2p/v1","definition":{"broker_addresses":["nats://broker.mysterium.network:4222","nats://broker.mysterium.network:4222","nats://51.158.129.204:4222"]}}],"quality":{"quality":0,"latency":0,"bandwidth":0}}}
Dec 01 22:33:34 ubuntu myst[1219162]: 2021-12-01T22:33:34.187 ERR core/discovery/discovery.go:208          > Failed to ping proposal error="failed to ping proposal: {0 service-proposal/v3 1 0x05b9004d70869f0c2e3da5f04d8229b89948b655 openvpn {NA CA Unknown 31798 DATACITY hosting} [{nats/p2p/v1 {[nats://broker.mysterium.network:4222 nats://broker.mysterium.network:4222 nats://51.158.129.204:4222]}}] <nil> {0 0 0}}: failed to send message 'signed.NZYt4_zSpfRzDl-x3KmpGfgrsVZ69Ky7xQyon9Q5CaMUAaBGPFVUxUb02RgdsJ5Qyfl6_64dX8rp3Aou3Cp11wA.1638416014.*.proposal-ping.v3': nats: connection closed"
Dec 01 22:33:34 ubuntu myst[1219162]: 2021-12-01T22:33:34.188 ERR core/discovery/discovery.go:208          > Failed to ping proposal error="failed to ping proposal: {0 service-proposal/v3 1 0x05b9004d70869f0c2e3da5f04d8229b89948b655 wireguard {NA CA Unknown 31798 DATACITY hosting} [{nats/p2p/v1 {[nats://broker.mysterium.network:4222 nats://broker.mysterium.network:4222 nats://51.158.129.204:4222]}}] <nil> {0 0 0}}: failed to send message 'signed.NZYt4_zSpfRzDl-x3KmpGfgrsVZ69Ky7xQyon9Q5CaMUAaBGPFVUxUb02RgdsJ5Qyfl6_64dX8rp3Aou3Cp11wA.1638416014.*.proposal-ping.v3': nats: connection closed"
Dec 01 22:36:34 ubuntu myst[1219162]: 2021-12-01T22:36:34.189 DBG communication/nats/sender.go:59          > Message "signed.Tv1QpYJR1YWxGMCauzHjSymG75DkGxBLTUKbwPZkNOpkhNohrsWxp1-qJYUbLi-45Vsp3C7jSOwQaOIGQRynVwE.1638416194.*.proposal-ping.v3" sending: {"proposal":{"id":0,"format":"service-proposal/v3","compatibility":1,"provider_id":"0x05b9004d70869f0c2e3da5f04d8229b89948b655","service_type":"openvpn","location":{"continent":"NA","country":"CA","city":"Unknown","asn":31798,"isp":"DATACITY","ip_type":"hosting"},"contacts":[{"type":"nats/p2p/v1","definition":{"broker_addresses":["nats://broker.mysterium.network:4222","nats://broker.mysterium.network:4222","nats://51.158.129.204:4222"]}}],"quality":{"quality":0,"latency":0,"bandwidth":0}}}
Dec 01 22:36:34 ubuntu myst[1219162]: 2021-12-01T22:36:34.190 ERR core/discovery/discovery.go:208          > Failed to ping proposal error="failed to ping proposal: {0 service-proposal/v3 1 0x05b9004d70869f0c2e3da5f04d8229b89948b655 openvpn {NA CA Unknown 31798 DATACITY hosting} [{nats/p2p/v1 {[nats://broker.mysterium.network:4222 nats://broker.mysterium.network:4222 nats://51.158.129.204:4222]}}] <nil> {0 0 0}}: failed to send message 'signed.Tv1QpYJR1YWxGMCauzHjSymG75DkGxBLTUKbwPZkNOpkhNohrsWxp1-qJYUbLi-45Vsp3C7jSOwQaOIGQRynVwE.1638416194.*.proposal-ping.v3': nats: connection closed"
Dec 01 22:36:34 ubuntu myst[1219162]: 2021-12-01T22:36:34.191 DBG communication/nats/sender.go:59          > Message "signed.Tv1QpYJR1YWxGMCauzHjSymG75DkGxBLTUKbwPZkNOpkhNohrsWxp1-qJYUbLi-45Vsp3C7jSOwQaOIGQRynVwE.1638416194.*.proposal-ping.v3" sending: {"proposal":{"id":0,"format":"service-proposal/v3","compatibility":1,"provider_id":"0x05b9004d70869f0c2e3da5f04d8229b89948b655","service_type":"wireguard","location":{"continent":"NA","country":"CA","city":"Unknown","asn":31798,"isp":"DATACITY","ip_type":"hosting"},"contacts":[{"type":"nats/p2p/v1","definition":{"broker_addresses":["nats://broker.mysterium.network:4222","nats://broker.mysterium.network:4222","nats://51.158.129.204:4222"]}}],"quality":{"quality":0,"latency":0,"bandwidth":0}}}
Dec 01 22:36:34 ubuntu myst[1219162]: 2021-12-01T22:36:34.191 ERR core/discovery/discovery.go:208          > Failed to ping proposal error="failed to ping proposal: {0 service-proposal/v3 1 0x05b9004d70869f0c2e3da5f04d8229b89948b655 wireguard {NA CA Unknown 31798 DATACITY hosting} [{nats/p2p/v1 {[nats://broker.mysterium.network:4222 nats://broker.mysterium.network:4222 nats://51.158.129.204:4222]}}] <nil> {0 0 0}}: failed to send message 'signed.Tv1QpYJR1YWxGMCauzHjSymG75DkGxBLTUKbwPZkNOpkhNohrsWxp1-qJYUbLi-45Vsp3C7jSOwQaOIGQRynVwE.1638416194.*.proposal-ping.v3': nats: connection closed"
Dec 01 22:39:34 ubuntu myst[1219162]: 2021-12-01T22:39:34.191 DBG communication/nats/sender.go:59          > Message "signed.GGBUBinVBgMfeIFrT7ssLG5f5XEG1PBm7xlsFvLz73pQVJj6iUcY75mG_brsOMb7CBf_PMSRr7zmyVm5LdrhxAE.1638416374.*.proposal-ping.v3" sending: {"proposal":{"id":0,"format":"service-proposal/v3","compatibility":1,"provider_id":"0x05b9004d70869f0c2e3da5f04d8229b89948b655","service_type":"openvpn","location":{"continent":"NA","country":"CA","city":"Unknown","asn":31798,"isp":"DATACITY","ip_type":"hosting"},"contacts":[{"type":"nats/p2p/v1","definition":{"broker_addresses":["nats://broker.mysterium.network:4222","nats://broker.mysterium.network:4222","nats://51.158.129.204:4222"]}}],"quality":{"quality":0,"latency":0,"bandwidth":0}}}
Dec 01 22:39:34 ubuntu myst[1219162]: 2021-12-01T22:39:34.193 ERR core/discovery/discovery.go:208          > Failed to ping proposal error="failed to ping proposal: {0 service-proposal/v3 1 0x05b9004d70869f0c2e3da5f04d8229b89948b655 openvpn {NA CA Unknown 31798 DATACITY hosting} [{nats/p2p/v1 {[nats://broker.mysterium.network:4222 nats://broker.mysterium.network:4222 nats://51.158.129.204:4222]}}] <nil> {0 0 0}}: failed to send message 'signed.GGBUBinVBgMfeIFrT7ssLG5f5XEG1PBm7xlsFvLz73pQVJj6iUcY75mG_brsOMb7CBf_PMSRr7zmyVm5LdrhxAE.1638416374.*.proposal-ping.v3': nats: connection closed"
Dec 01 22:39:34 ubuntu myst[1219162]: 2021-12-01T22:39:34.194 DBG communication/nats/sender.go:59          > Message "signed.GGBUBinVBgMfeIFrT7ssLG5f5XEG1PBm7xlsFvLz73pQVJj6iUcY75mG_brsOMb7CBf_PMSRr7zmyVm5LdrhxAE.1638416374.*.proposal-ping.v3" sending: {"proposal":{"id":0,"format":"service-proposal/v3","compatibility":1,"provider_id":"0x05b9004d70869f0c2e3da5f04d8229b89948b655","service_type":"wireguard","location":{"continent":"NA","country":"CA","city":"Unknown","asn":31798,"isp":"DATACITY","ip_type":"hosting"},"contacts":[{"type":"nats/p2p/v1","definition":{"broker_addresses":["nats://broker.mysterium.network:4222","nats://broker.mysterium.network:4222","nats://51.158.129.204:4222"]}}],"quality":{"quality":0,"latency":0,"bandwidth":0}}}
Dec 01 22:39:34 ubuntu myst[1219162]: 2021-12-01T22:39:34.194 ERR core/discovery/discovery.go:208          > Failed to ping proposal error="failed to ping proposal: {0 service-proposal/v3 1 0x05b9004d70869f0c2e3da5f04d8229b89948b655 wireguard {NA CA Unknown 31798 DATACITY hosting} [{nats/p2p/v1 {[nats://broker.mysterium.network:4222 nats://broker.mysterium.network:4222 nats://51.158.129.204:4222]}}] <nil> {0 0 0}}: failed to send message 'signed.GGBUBinVBgMfeIFrT7ssLG5f5XEG1PBm7xlsFvLz73pQVJj6iUcY75mG_brsOMb7CBf_PMSRr7zmyVm5LdrhxAE.1638416374.*.proposal-ping.v3': nats: connection closed"
```
[log.gz](https://github.com/mysteriumnetwork/node/files/7638830/log.gz)

**Comment by chompomonim** [comment]
Can't reproduce on latest version. Please open new issue with more context when you'll know how to reproduce.

**Comment by Snawoot** [comment]
Fixed via mysteriumnetwork/nats-proxy#13
